ffi/flake.nix

```diff
@@ -0,0 +1,119 @@
+{
+  # To use:
+  #  - Install nix: https://github.com/DeterminateSystems/nix-installer?tab=readme-ov-file#install-nix
+  #  - Run from ffi/ dir: `nix build .#firewood-ffi`
+  #  - Run from anywhere: `nix develop 'github:ava-labs/firewood?dir=ffi&ref=[SHA]'`
+
+  description = "Firewood FFI library and development environment";
+
+  inputs = {
+    nixpkgs.url = "https://flakehub.com/f/NixOS/nixpkgs/0.2505.*.tar.gz";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    crane.url = "github:ipetkov/crane";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, rust-overlay, crane, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+    let
+      overlays = [ (import rust-overlay) ];
+      pkgs = import nixpkgs { inherit system overlays; };
+      inherit (pkgs) lib;
+
+      rustToolchain = pkgs.rust-bin.stable.latest.default.override {
+        extensions = [ "rust-src" "rustfmt" "clippy" ];
+      };
+
+      craneLib = (crane.mkLib pkgs).overrideToolchain rustToolchain;
+
+      # Extract crate info from Cargo.toml files
+      ffiCargoToml = builtins.fromTOML (builtins.readFile ./Cargo.toml);
+      workspaceCargoToml = builtins.fromTOML (builtins.readFile ../Cargo.toml);
+
+      src = lib.cleanSourceWith {
+        src = craneLib.path ./..;
+        filter = path: type:
+          (lib.hasSuffix "\.md" path) ||
+          (craneLib.filterCargoSources path type);
+      };
+
+      commonArgs = {
+        inherit src;
+        strictDeps = true;
+        dontStrip = true;
+
+        # Build only the firewood-ffi crate
+        pname = ffiCargoToml.package.name;
+        version = workspaceCargoToml.workspace.package.version;
+
+        CARGO_PROFILE = "maxperf";
+
+        nativeBuildInputs = with pkgs; [
+          pkg-config
+        ] ++ lib.optionals stdenv.isDarwin [
+          darwin.apple_sdk.frameworks.Security
+          darwin.apple_sdk.frameworks.SystemConfiguration
+        ];
+      };
+
+      cargoArtifacts = craneLib.buildDepsOnly (commonArgs // {
+        cargoExtraArgs = "--package ${ffiCargoToml.package.name} --features ethhash,logger";
+      });
+
+      firewood-ffi = craneLib.buildPackage (commonArgs // {
+        inherit cargoArtifacts;
+        cargoExtraArgs = "--package ${ffiCargoToml.package.name} --features ethhash,logger --frozen";
+
+        # Install the static library and header
+        postInstall = ''
+          mkdir -p $out/lib $out/include
+          cp target/*/libfirewood_ffi.a $out/lib/ || cp target/release/libfirewood_ffi.a $out/lib/
```

Comment:
the `*` should probably be replaced with `maxperf` as the target directory is deterministic on the profile used. If the resulting artifact isn't in the `maxperf` directory, then we're using the wrong profile.